impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterTestInvocationContextProvider.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.testing.kafka.common;
+
+import org.junit.jupiter.api.extension.AfterTestExecutionCallback;
+import org.junit.jupiter.api.extension.BeforeTestExecutionCallback;
+import org.junit.jupiter.api.extension.Extension;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContext;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContextProvider;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static java.util.Arrays.asList;
+
+public class KafkaClusterTestInvocationContextProvider implements TestTemplateInvocationContextProvider {
```

Comment:
Why is this a separate thing from the existing extension? In particular, unless it is part of the existing extension I don't think it would be possible to write `@TestTemplate` methods which test the Cartesian product of a range of versions and `{kraft, zookeeper}`, or `{combined, dedicated}` KRaft nodes, or `{with SSL, without SSL}`.

Reply:
done